### PR TITLE
Making Uri proto, microuri support mandatory and longuri support optional

### PR DESCRIPTION
**Pull request by ruchirchauhan**
This PR implements issue #102 

It modifies uri.proto:

1. UEntity name is now optional
2. UResource name is now optional

### CODE REVIEW

File: up-core-api/uprotocol/uri.proto [hunk 1]
```diff
@@ -74,8 +74,8 @@ message UAuthority {
 // A uE that publishes events is a <b>Service</b> role.<br>
 // A uE that consumes events is an <b>Application</b> role.
 message UEntity {
-  string name = 1;    // Name of the entity
-  optional uint32 id = 2;      // The numeric ID for the uEntity assigned from the name/number registry
+  optional string name = 1;          // Name of the entity
+  uint32 id = 2;                     // The numeric ID for the uEntity assigned from the name/number registry
```
Comment (stevenhartley) [inline up-core-api/uprotocol/uri.proto:78]:
```suggestion
  optional uint32 id = 2;                     // The numeric ID for the uEntity assigned from the name/number registry
```

File: up-core-api/uprotocol/uri.proto [hunk 2]
```diff
@@ -91,10 +91,10 @@ message UEntity {
 // Resources are unique when prepended with UAuthority that represents the device and
 // UEntity that represents the service.
 message UResource {
-  string name = 1;      // Name of the resource
+  optional string name = 1;      // Name of the resource
   optional string instance = 2;  // Instance of the resource
   optional string message = 3;   // Message type for the resource
-  optional uint32 id = 4;        // Numerical representation of the UResource portion of the UUri
+  uint32 id = 4;                 // Numerical representation of the UResource portion of the UUri
```
Comment (stevenhartley) [inline up-core-api/uprotocol/uri.proto:97]:
```suggestion
  optional uint32 id = 4;                 // Numerical representation of the UResource portion of the UUri
```

**Review by stevenhartley — CHANGES_REQUESTED** [review]
The purpose of this change is to allow us to build URIs that contain only numbers that could be used for micro and short URI format however if we put id as mandatory and name optional, we will not be able to build URIs that are in long format only (meaning URIs that only contain names). Since UUri is a container to store the URI information, it needs to support the ability to contain in its belly:

1. Names only
2. Numbers only
3. Names & numbers

The changes I've proposed allows the UUri to contain any of the above information, please consider the suggestions

File: up-core-api/uprotocol/uri.proto (hunk 1 above)
Comment (ruchirchauhan) [reply up-core-api/uprotocol/uri.proto]: Done

**Review by stevenhartley — APPROVED** [review]
Approved

**Comment by sophokles73** [comment]
@stevenhartley it would be nice if we could squash PRs like this before merging in the future to keep the commit log in a clean and consistent state. We now have three commits in the log, each of which represents intermediate work and which only make sense when applied together ...

**Pull request merged**